### PR TITLE
To fix the chainability at epoch zero for some schedulers

### LINKED ISSUE #4281: Update reference scripts to use the "Batteries Included" utils
## 🚀 Feature
As part of the "Batteries Included" initiative (#3911) we are adding a number of new utils that can be used to produce SOTA results. Once those utils are landed, we should update our reference scripts to use them. 

More specifically we need to:
- [ ] Update the reference scripts to use PyTorch's new [linear warmup](https://github.com/pytorch/pytorch/pull/60836) scheduler.
  - The reference scripts should be updated to let users define the `warmup_method` (if any), the `warmup_iters` and the `warmup_factor`. The warmup scheduler should be chained with other existing schedulers.
  - The Object Detection recipe needs to be BC compatible and replace our [custom linear warmup](https://github.com/pytorch/vision/blob/dc42f933f3343c76727dbfba6e4242f1bcb8e1a0/references/detection/engine.py#L20-L24) approach with the one from PyTorch.
  - Similarly the Video Classification recipe needs also to be BC compatible and replace our [customer scheduler](https://github.com/pytorch/vision/blob/master/references/video_classification/scheduler.py#L5).
  - All other recipes (Classification and Segmentation), should be updated to optionally use warm up. The addition should be BC compatible and turned off by default.
- [x] Update the Classification reference to use PyTorch's new label smoothing implementation.
  - We will support the case of label smoothing by using the upcoming `label_smoothing` argument of CrossEntropyLoss (https://github.com/pytorch/pytorch/pull/63122). 
  - A new `label_smoothing` argument should be introduced on the [script](https://github.com/pytorch/vision/blob/c2ab0c59f42babf9ad01aa616cd8a901daac86dd/references/classification/train.py#L165). To ensure BC, its default value should be 0.0 (no label smoothing).

### PR DESCRIPTION
**Pull request by iramazanli**
It has been discussed in the https://github.com/pytorch/pytorch/pull/60836#issuecomment-899084092 that we have observed an obstacle to chain some type of learning rate schedulers. In particular we observed

* some of the learning rate schedulers returns initial learning rates at epoch 0 as
```
       return self.base_lrs`
```

* This can be a problem when two schedulers called as chained as

```
     scheduler1.step()
     scheduler2.step()
```

in particular, we completely ignore the effect of scheduler1 at epoch 0.  This could not be an issue if at epoch 0, scheduler1 was ineffective as in many schedulers, however for schedulers as WarmUp Schedulers, where at epoch 0 schedulers multiplicative value is smaller than 1 this could lead to undesired behaviors. 

The following code snippet illustrates the problem better

## Reproducing the bug

```python
import torch
from torch.nn import Parameter
from torch.optim import SGD
from torch.optim.lr_scheduler import WarmUpLR, ExponentialLR

model = [Parameter(torch.randn(2, 2, requires_grad=True))]
optimizer = SGD(model, 1.0)
scheduler1 = WarmUpLR(optimizer, warmup_factor=0.1, warmup_iters=5, warmup_method="constant")
scheduler2 = ExponentialLR(optimizer, gamma=0.9)

for epoch in range(10):
     print(epoch, scheduler2.get_last_lr()[0])
     optimizer.step()
     scheduler1.step()
     scheduler2.step()
```

### Current Result

``` 
0 1.0
1 0.9
2 0.81
3 0.7290000000000001
4 0.6561000000000001
5 5.904900000000001
6 5.314410000000001
7 4.782969000000001
8 4.304672100000001
9 3.874204890000001
```

### Expected Result

``` 
0 1.0
1 0.9
2 0.81
3 0.7290000000000001
4 0.6561000000000001
5 0.5904900000000001
6 0.5314410000000001
7 0.4782969000000001
8 0.4304672100000001
9 0.3874204890000001
```

Partially resolves https://github.com/pytorch/vision/issues/4281

### CODE REVIEW

**Review by datumbox — APPROVED** [review]
LGTM, thanks @iramazanli for fixing this so quickly.

I tested your patch on latest nightly with a slightly modified loop:
```python
for epoch in range(10):
     print(epoch, scheduler2.get_lr())
     optimizer.step()
     scheduler1.step()
     scheduler2.step()
```

And I get the expected result:
```
0 [0.1]
1 [0.08100000000000002]
2 [0.07290000000000002]
3 [0.06561000000000002]
4 [0.05904900000000002]
5 [0.5314410000000002]
6 [0.47829690000000014]
7 [0.43046721000000016]
8 [0.38742048900000015]
9 [0.34867844010000015]
```

Which is the combined effect of both schedulers.

**Review by fmassa — APPROVED** [review]
Great, thanks for fixing this!

**Comment by iramazanli** [comment]
> LGTM, thanks @iramazanli for fixing this so quickly.
> 
> I tested your patch on latest nightly with a slightly modified loop:
> 
> ```python
> for epoch in range(10):
>      print(epoch, scheduler2.get_lr())
>      optimizer.step()
>      scheduler1.step()
>      scheduler2.step()
> ```
> 
> And I get the expected result:
> 
> ```
> 0 [0.1]
> 1 [0.08100000000000002]
> 2 [0.07290000000000002]
> 3 [0.06561000000000002]
> 4 [0.05904900000000002]
> 5 [0.5314410000000002]
> 6 [0.47829690000000014]
> 7 [0.43046721000000016]
> 8 [0.38742048900000015]
> 9 [0.34867844010000015]
> ```
> 
> Which is the combined effect of both schedulers.

That's amazing! lets merge this PR then :)
